### PR TITLE
Containerized builds and release process

### LINKED ISSUE #4: Add a container build of calico-bgp-daemon
And update the calico/node make target to extract the binary from the container

### PR DESCRIPTION
**Pull request by gunjan5**
Fixes #4 
Fixes https://github.com/projectcalico/calicoctl/issues/1173

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,14 +11,52 @@ vendor:
 	glide install -strip-vcs -strip-vendor --cache; \
 	chown $(shell id -u):$(shell id -u) -R vendor'
 
-calico-bgp-daemon: main.go vendor
+binary: dist/gobgpd
+
+dist/gobgp:
+	mkdir -p $(@D)
+	docker run --rm -v `pwd`/dist:/code golang:1.7 \
```
Comment (tomdee) [inline Makefile]: Why not use calico/go-build here too?

**Comment by gunjan5** [comment]
@tomdee when I switch from `golang:1.7` to `calico/go-build` container, it seems to fail on Semaphore with 
```
Starting with UID : 9001
cp: can't create '/go/code/gobgp': Permission denied
```
(works fine on my mac) 

I think it might have something to do with the UID stuff we have at https://github.com/projectcalico/go-build/blob/master/entrypoint.sh#L7

Any idea how to fix that without having to modify the go-build container entrypoint?

**Comment by tomdee** [comment]
Have you looked at how the go-build container is used in the libnetwork plugin repo? You need to pass in an env car to set the right user

**Comment by gunjan5** [comment]
@tomdee yup I was missing that. Fixed it. PTAL

**Comment by heschlie** [comment]
LGTM!

**Pull request merged**